app/javascript/components/admin/UserModal.js

```diff
@@ -17,31 +22,51 @@ class UserModal extends React.Component {
       roleTitle: this.props.initialUserData.role_title || this.props.roles[0],
       isAPIEnabled: this.props.initialUserData.is_api_enabled || false,
       isLocked: this.props.initialUserData.is_locked || false,
+      lockReason: this.props.initialUserData.lock_reason || '',
+      autoLockReason: this.props.initialUserData.auto_lock_reason || '',
+      activeState: this.props.initialUserData.active_state,
+      status: this.props.initialUserData.status,
       notes: this.props.initialUserData.notes || '',
+      lockReasonOptions:
+        this.props.initialUserData.lock_reason === 'Auto-locked by the System'
+          ? LOCK_REASON_OPTIONS.concat(['Auto-locked by the System']).sort()
+          : LOCK_REASON_OPTIONS,
+      days_since_last_login_for_inactivity: '',
     };
   }
 
-  handleChange = event => {
-    this.setState({ [event.target.name]: event?.target?.value || '' });
-  };
+  /**
+   * Gets the days_since_last_login_for_inactivity value for display in tooltip via an axios GET request.
+   */
+  getDaysSinceLastLoginForInactivity() {
+    axios.get(window.BASE_PATH + '/admin/days_since_last_login_for_inactivity').then(response => {
+      this.setState({ days_since_last_login_for_inactivity: response.data.days_since_last_login_for_inactivity });
+    });
+  }
 
-  handleJurisdictionChange = data => {
-    this.setState({ jurisdiction_path: data.value });
+  handleChange = event => {
+    const value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
+    this.setState({ [event.target.name]: value });
   };
 
-  handleRoleChange = data => {
-    this.setState({ roleTitle: data.value });
+  getStatusValue = () => {
+    if (this.state.isLocked) {
+      return this.state.lockReason ? { label: this.state.lockReason, value: this.state.lockReason } : { label: '', value: '' };
+    } else {
+      return { label: this.state.activeState, value: this.state.activeState };
+    }
   };
 
-  handleLockedStatusChange = event => {
-    const val = event.target.checked;
-    this.setState({ isLocked: val });
+  getTooltipText = () => {
+    const activeStatusText = `Logged into the system within the last ${this.state.days_since_last_login_for_inactivity} days`;
```

Comment:
I'm seeing this value come back null:

<img width="537" alt="Screen Shot 2022-01-14 at 8 51 49 AM" src="https://user-images.githubusercontent.com/35042815/149526032-86ecca1c-2354-4a9a-a101-b770e0924a7c.png">

<img width="495" alt="Screen Shot 2022-01-14 at 8 48 55 AM" src="https://user-images.githubusercontent.com/35042815/149526098-ab4c8723-c840-45ef-a987-a6483409d393.png">

Reply:
[15b93cb85d43b65216cde511f24c323941562ae6](https://github.com/SaraAlert/SaraAlert/pull/1180/commits/15b93cb85d43b65216cde511f24c323941562ae6)